### DISCUSSION TITLE
Enhancement: Make a keyboard layout detection

### DISCUSSION THREAD

**Discussion by Midou36O** [FunkinCrew/Funkin#1587] [Ideas]
## What is your suggestion, and why should it be implemented?
**The Ability to detect what keyboard layout you have**, as some people (like me) would like to use WASD controls but seems to be a bit garbled (Azerty layout for example puts the Z on the W and the A on the Q), looking at the code source i can't seem to find keyboard letter bindings (or i just didn't search too much :-/ ) 

Qwerty layout with WASD:
![image](https://user-images.githubusercontent.com/45198486/106872558-bb36de80-66d3-11eb-959a-cf126b695c97.png)

Azerty layout with WASD:
![image](https://user-images.githubusercontent.com/45198486/106872935-2bddfb00-66d4-11eb-92a1-e8826af4f17e.png) 
(you can see it's hard to play like that)

**Comment by regulad** [comment]
Rebindable keys are semi-implemented right now, there is a file named `controls.txt` in `assets/data` that should allow you to rebind keys.

It's still a little broken, but worth trying out.

**Comment by Avalance112** [comment]
im trying to change it's but won't work

**Comment by regulad** [comment]
> 
> 
> im trying to change it's but won't work

>It's still a little broken, but worth trying out.

**Comment by regulad** [comment]
Hadn't tested it myself until now. Very broken.

**Comment by Injourn** [comment]
> Rebindable keys are semi-implemented right now, there is a file named controls.txt in assets/data that should allow you to rebind keys.

>It's still a little broken, but worth trying out.

I've made an attempt to create a pull request for this specific feature, #301, However, the options part of the menu has been removed.

**Comment by PKPenguin321** [comment]
@regulad That file is nonfunctional in the live build and does nothing, wait for #301 to be added

**Comment by regulad** [comment]
> @regulad That file is nonfunctional in the live build and does nothing, wait for #301 to be added

I realized lol
